### PR TITLE
spotube: use `makeBinaryWrapper`

### PR DESCRIPTION
**Pull request by Guanran928**
## Description of changes

https://github.com/NixOS/nixpkgs/pull/297751#issuecomment-2013027739
Or `nix run nixpkgs#spotube` would be stuck forever.
```
❯ nix run nixpkgs#spotube
^C⏎

nixpkgs on  spotube took 4m53s
```

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by wegank — APPROVED** [review]

**Comment by eclairevoyant** [comment]
For my understanding, is [this bit of nixpkgs docs](https://nixos.org/manual/nixpkgs/unstable/#makewrapper-executable-wrapperfile-args) still accurate, or has the darwin situation changed?

> Using the `makeBinaryWrapper` implementation is usually preferred, as it creates a tiny _compiled_ wrapper executable, that can be used as a shebang interpreter. This is needed mostly on Darwin, where shebangs cannot point to scripts, [due to a limitation with the `execve`-syscall](https://stackoverflow.com/questions/67100831/macos-shebang-with-absolute-path-not-working).

**Review by drupol — CHANGES_REQUESTED** [review]
Please use makeBinaryWrapper

**Comment by drupol** [comment]
They do almost the same things. But when it comes to Darwin, it's safer to use makeBinaryWrapper.

**Comment by drupol** [comment]
It's still makeWrapper.

**Review by drupol — APPROVED** [review]

**Review by drupol — CHANGES_REQUESTED** [review]
There's another instance of `makeWrapper` down in the file.

**Review by drupol — CHANGES_REQUESTED** [review]
Please update the commit log as such:

```
spotube: use `makeBinaryWrapper`
```

File: pkgs/by-name/sp/spotube/package.nix [hunk 1]
```diff
@@ -73,7 +73,7 @@ let
     nativeBuildInputs = [
       autoPatchelfHook
       dpkg
-      makeWrapper
+      makeBinaryWrapper
```
Comment (wegank) [inline pkgs/by-name/sp/spotube/package.nix:76]: Doesn't `wrapGAppsHook` already provide `makeBinaryWrapper`?

File: pkgs/by-name/sp/spotube/package.nix [hunk 2]
```diff
@@ -51,13 +51,13 @@ let
 
     sourceRoot = ".";
 
-    nativeBuildInputs = [ undmg ];
+    nativeBuildInputs = [ undmg wrapGAppsHook ];
```
Comment (drupol) [inline pkgs/by-name/sp/spotube/package.nix:54]: Why is this added ?

Comment (drupol) [reply pkgs/by-name/sp/spotube/package.nix:54]: Ok, but what's the reason to use `wrapGAppsHook` over `makeBinaryWrapper` ?

**Comment by eclairevoyant** [comment]
> https://github.com/NixOS/nixpkgs/blob/0f150faf98c9e5d9709bfb63fc9c23f45399baef/pkgs/build-support/setup-hooks/make-binary-wrapper/make-binary-wrapper.sh#L37
> 
> I believe makeWrapper and makeBinaryWrapper are the same

It depends on which hook was added to `nativeBuildInputs`, and you can see by the filename (`make-binary-wrapper.sh`) that this you linked the `makeBinaryWrapper` implementation of the bash function `makeWrapper`, while what you had written originally in the PR was using the `makeWrapper` impl of the bash function `makeWrapper`, which is different:

https://github.com/NixOS/nixpkgs/blob/d44b87d86f64b756e65cbf89506ca738a7204788/pkgs/build-support/setup-hooks/make-wrapper.sh#L36

File: pkgs/by-name/sp/spotube/package.nix (hunk 2 above)
Comment (eclairevoyant) [reply pkgs/by-name/sp/spotube/package.nix:54]: It's a GTK app, not sure if that needs the special wrapping on darwin. But as written I believe this would get double-wrapped, since `dontWrapGApps` would be false by default.

Comment (wegank) [reply pkgs/by-name/sp/spotube/package.nix:54]: > [#297790 (comment)](https://github.com/NixOS/nixpkgs/pull/297790#discussion_r1535277410)

But there was no `wrapGAppsHook` in `nativeBuildInputs` here.

**Review by drupol — APPROVED** [review]
